pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -461,55 +462,64 @@ func (h *upgradeHandler) cleanup(upgrade *harvesterv1.Upgrade, cleanJobs bool) e
 			}
 			jobs, err := h.jobCache.List(plan.Namespace, set.AsSelector())
 			if err != nil {
-				return err
+				return nil, err
```

Comment:
not specifically related to this PR,

but it deserves to skip the `IsNotFound` error

Reply:
I fixed the the suggestion below of `XXXClient.Delete` -> Check deletionTimestamp and IsNotFound error before deleting.

For this line though, would we get `IsNotFound` error while listing? I think we don't need to handle it if not.
> 			jobs, err := h.jobCache.List(plan.Namespace, set.AsSelector())